DNN Platform/Library/Entities/Controllers/HostController.cs

```diff
@@ -20,49 +19,27 @@ namespace DotNetNuke.Entities.Controllers
     using DotNetNuke.Services.Log.EventLog;
     using DotNetNuke.Web.Client;
 
-    /// <summary>
-    /// HostController provides business layer of host settings.
-    /// </summary>
-    /// <example>
-    /// <code lang="C#">
-    /// public static bool CheckUpgrade
-    /// {
-    ///     get
-    ///     {
-    ///         return HostController.Instance.GetBoolean("CheckUpgrade", true);
-    ///     }
-    /// }
-    /// </code>
-    /// </example>
-    public class HostController : ComponentBase<IHostController, HostController>, IHostController
+    using INewHostController = DotNetNuke.Abstractions.Entities.Controllers.IHostController;

```

Comment:
Any particular reason for this, rather than a traditional using?

Reply:
## Short Answer
In the deprecated partial class you will need to use the new abstractions interface. But now we have 2 interfaces with the exact same name. We can either alias the using statement as I did here, or use long fully qualified names.

## Long Answer
When I looked through the original history of the controller interfaces I discovered that they were originally built as a way to add more unit tests to DNN not ever designed for production use. The comment isn't shared across all controller interfaces. In addition to we need to extract an interface from our controller and move it over to the abstractions project so it is truly ready for dependency injection, as well as any future vNext of DNN.
https://github.com/dnnsoftware/Dnn.Platform/blob/b098a16d467ae3890ed13d654be092e4b5c0bad3/DNN%20Platform/Library/Entities/Portals/IPortalController.cs#L14-L18

My technique to organizing the code was to break it into a few sections
- Create interface in abstractions project
- Create partial class of controller implementation in Obsolete folder
- Update original class to partial class
- Remove deprecated inheritance to obsolete class (ServiceLocator and old interface)
- Add new interface to the original implementation
- **In the deprecated partial class you will need to use the new abstractions interface. But now we have 2 interfaces with the exact same name. We can either alias the using statement as I did here, or use long fully qualified names.**
  - All deprecated methods should use the new implementation and there shouldn't be 2 copies of the same implementation

Reply:
@bdukes and I chatted about this and decided it would be best to name the new interface `IHostSettingsService`. I have updated the PR and there are no longer any using aliases. I am going to mark this conversation as resolved.

https://github.com/dnnsoftware/Dnn.Platform/pull/3990#discussion_r474019228